idseq_dag/steps/run_srst2.py

```diff
@@ -0,0 +1,110 @@
+            command.execute(f"cp empty_file.txt {self.output_files_local()[2]}")
+            command.execute(f"cp empty_file.txt {self.output_files_local()[3]}")
+            command.execute(f"cp empty_file.txt {self.output_files_local()[4]}")
+        else:
+            # Post processing of amr data
+            results_full = os.path.join(self.output_dir_local, 'output__fullgenes__ARGannot_r2__results.txt')
+            results_full_dest = self.output_files_local()[2]
+            PipelineStepRunSRST2.mv_to_dest(results_full, results_full_dest)
+            self.process_amr_results(results_full_dest)
+
+    # Inherited method
+    def count_reads(self):
+        pass
+
+    @staticmethod
+    def mv_to_dest(src, dest):
+        """Moves src to dest path. 
+           Handles case where src and dest file are named the same and a mv "same file" error is thrown."""
+        if src == dest:
+            command.execute(f"mv {src} 'src'")
+            command.execute(f"mv 'src' {dest}")   
+        else:
+            command.execute(f"mv {src} {dest}")
+            
+    @staticmethod
+    def _get_pre_proc_amr_results(amr_raw_path):
+        """ Reads in raw amr results file outputted by srst2, and does initial processing of marking gene family."""
+        amr_results = pd.read_csv(amr_raw_path, delimiter="\t")
+        # Parse out gene family as substring after '_', e.g. Aph_AGly's gene family would be AGly
+        amr_results['gene_family'] = amr_results.apply(lambda row: row.gene.split('_', 1)[1], axis=1)
+        return amr_results
+
+    @staticmethod
+    def _summarize_amr_gene_families(amr_raw_path):
+        """Returns a gene family-level summary of total_genes_hit, total_coverage, and total_depth."""
+        amr_results = PipelineStepRunSRST2._get_pre_proc_amr_results(amr_raw_path)
+        amr_summary = amr_results.groupby(['gene_family']).agg({'gene_family': ['size'],'coverage':['sum'], 'depth':['sum']})
+        amr_summary.columns = [' '.join(col) for col in amr_summary.columns]
+        amr_summary = amr_summary.rename(columns={'gene_family size': 'total_gene_hits', 'coverage sum': 'total_coverage', 'depth sum': 'total_depth'}).reset_index()
+        return amr_summary
+
+
+    def process_amr_results(self, amr_results_path):
+        """ Writes processed amr result table with total_genes_hit, total_coverage,
+            and total_depth column values filled in for all genes to output files; and likewise with
+            the gene-family level summary of amr results table. """
+        amr_summary = PipelineStepRunSRST2._summarize_amr_gene_families(amr_results_path)
+        amr_summary.to_csv(
+            'amr_summary_results.csv',
+            mode='w',
+            index=False,
+            encoding='utf-8')
+        command.execute(f"mv amr_summary_results.csv {self.output_files_local()[4]}")
+        amr_results = PipelineStepRunSRST2._get_pre_proc_amr_results(amr_results_path)
```

Comment:
You're doing the same computation (`_get_pre_proc_amr_results`) twice on the same data -- once as part of `_summarize_amr_gene_families` and then again here. I would refactor so that you only do it once. I'm not sure `_get_pre_proc_amr_results` and `_summarize_amr_gene_families` need to be their own functions so you could streamline by moving them into `process_amr_results`, using comments to explaining what each block does.

Reply:
thanks for all your comments! addressed them in updates to this PR